lib/primitive/lib/Cardano/Wallet/Primitive/Types/AssetName.hs

```diff
@@ -59,7 +63,7 @@ import qualified Data.Text.Encoding as T
 newtype AssetName =
     -- | Construct an 'AssetName' without any validation.
     UnsafeAssetName { unAssetName :: ByteString }
-    deriving stock (Eq, Ord, Generic)
+    deriving stock (Data, Eq, Ord, Generic)
```

Comment:
why you need this instance? For `gfoldl` ?

Reply:
> why you need this instance? For `gfoldl` ?

Ah good question!

This is because I'd like to include values of type `ApiWalletAssets` within constructors of `ApiError`. But `ApiError` is required to be an instance of `Data`.

Reply:
aha, the requirement that one can see only when compiling - hehe. :+1:

Reply:
Indeed!

Though now that you mention it, it would be nice if we had a way to identify all redundant instances of things like `Generic`, `Data`, and `Typeable`. I'm guessing that these instances can take a while to compile.